dipy/core/geometry.py

```diff
@@ -906,3 +910,102 @@ def compose_transformations(*mats):
 
     return prev
 
+
+def perpendicular_directions(v, num=30, half=False):
+    r""" Computes n evenly spaced perpendicular directions relative to a given
+    vector v
+
+    Parameters
+    -----------
+    v : array (3,)
+        Array containing the three cartesian coordinates of vector v
+    num : int, optional
+        Number of perpendicular directions to generate
+    half : bool, optional
+        If half is True, perpendicular directions are sampled on halp of the
+        unit circumference perpendicular to v, otherwive perpendicular
+        directions are sampled on the full circumference. Default of half is
+        False
+
+    Returns
+    -------
+    psamples : array (n, 3)
+        array of vectors perpendicular to v
+
+    Notes
+    --------
+    Perpendicular directions are estimated using the following two step
+    procedure:
+    
+        1) the perpendicular directions are first sampled in a unit
+        circumference parallel to the plane normal to the x-axis.
+
+        2) Samples are then rotated and aligned to the plane normal to vector
+        v. The rotational matrix for this rotation is constructed as reference
+        frame basis which axis are the following:
+            - The first axis is vector v
+            - The second axis is defined as the normalized vector given by the
+            cross product between vector v and the unit vector aligned to the
+            x-axis
+            - The third axis is defined as the cross product between the
+            previous computed vector and vector v.
+
+    Following this two steps, coordinates of the final perpendicular directions
+    are given as:
+
+    .. math::
+
+        \left [ -\sin(a_{i}) \sqrt{{v_{y}}^{2}+{v_{z}}^{2}}
+        \; , \;
+        \frac{v_{x}v_{y}\sin(a_{i})-v_{z}\cos(a_{i})}
+        {\sqrt{{v_{y}}^{2}+{v_{z}}^{2}}}
+        \; , \;
+        \frac{v_{x}v_{z}\sin(a_{i})-v_{y}\cos(a_{i})}
+        {\sqrt{{v_{y}}^{2}+{v_{z}}^{2}}} \right  ]
+
+    This procedure has a singularity when vector v is aligned to the x-axis. To
+    solve this singularity, perpendicular directions in procedure's step 1 are
+    defined in the plane normal to y-axis and the second axis of the rotated
+    frame of reference is computed as the normalized vector given by the cross
+    product between vector v and the unit vector aligned to the y-axis. 
+    Following this, the coordinates of the perpendicular directions are given
+    as:
+
+        \left [ -\frac{\left (v_{x}v_{y}\sin(a_{i})+v_{z}\cos(a_{i}) \right )}
+        {\sqrt{{v_{x}}^{2}+{v_{z}}^{2}}}
+        \; , \;
+        \sin(a_{i}) \sqrt{{v_{x}}^{2}+{v_{z}}^{2}}
+        \; , \;
+        \frac{v_{y}v_{z}\sin(a_{i})+v_{x}\cos(a_{i})}
+        {\sqrt{{v_{x}}^{2}+{v_{z}}^{2}}} \right  ]
+
+    For more details see:
+
+    http://gsoc2015dipydki.blogspot.it/2015/07/rnh-post-8-computing-perpendicul
+    ar.html
```

Comment:
Please use the rst link syntax here. http://docutils.sourceforge.net/docs/user/rst/quickref.html#external-hyperlink-targets